frontend/src/ui/player/PlayerGroupContext.tsx

```diff
@@ -0,0 +1,40 @@
+import { Paella } from "paella-core";
+import React, {
+    createContext,
+    useContext,
+    useRef,
+    PropsWithChildren,
+    MutableRefObject,
+} from "react";
+
+type PlayerGroupContext = {
+    players: MutableRefObject<Paella[]>;
```

Comment:
Oh you're right, I don't think this needs a `ref` at all. At least I can't see any difference in the number of renderings. The list is "initialized" when the page is loaded and doesn't change after that, so it should be fine with `players` just being a `Set`.

Reply:
I would have assumed that `useRef` is necessary here. Otherwise this set is re-created (empty) whenever that provider is rendered. But I couldn't find a way to make this lead to problems. So maybe it's fine. I'm a bit uncomfortable merging this but it works so :shrug: